idea/src/org/jetbrains/kotlin/idea/intentions/branchedTransformations/branchedTransformationUtils.kt

```diff
@@ -68,7 +71,11 @@ private fun KtExpression?.getWhenConditionSubjectCandidate(): KtExpression? = wh
         val lhs = left
         when (operationToken) {
             KtTokens.IN_KEYWORD, KtTokens.NOT_IN -> lhs
-            KtTokens.EQEQ -> lhs as? KtNameReferenceExpression ?: right
+            KtTokens.EQEQ -> {
+                lhs as? KtNameReferenceExpression
+                    ?: (lhs as? KtQualifiedExpression)?.takeIf { it.selectorExpression is KtNameReferenceExpression }
```

Comment:
It's better to apply here the same checks for the right part, if the left one does not satisfy our conditions.